presentation/src/main/java/org/sagebionetworks/research/presentation/recorder/service/RecorderManager.java

```diff
@@ -137,19 +137,22 @@ public void onServiceConnected(final ComponentName componentName, final IBinder
                 }
             }
 
-            for (Recorder recorder : this.getActiveRecorders().values()) {
-                this.compositeDisposable.add(
-                        ((Recorder<? extends Result>) recorder).getResult()
-                                .subscribe((result) -> {
-                                    for (ObservableEmitter<Result> emitter : this.observers) {
-                                        emitter.onNext(result);
-                                        emitter.onComplete();
-                                    }
-                                }, (throwable -> {
-                                    for (ObservableEmitter<Result> emitter : this.observers) {
-                                        emitter.onError(throwable);
-                                    }
-                                })));
+            activeRecorders = this.getActiveRecorders();
```

Comment:
I think I had the same change. For me it was happening when the service wasn't ready by the time the first step runs. I want to add something like a Single<Binder> to prevent this

Reply:
I think it should be a Maybe since there aren't necessarily any recorders.

Reply:
Oh, I meant I did something like Single<RecorderManager> that returns after the service is bound. Haven't figured out how to handle unbind